Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -330,12 +361,19 @@ private protected async Task<bool> CompletesWithinTimeoutAsync(Task target, Time
         Task completedTask =
             await Task.WhenAny(target, Clock.DelayAsync(remainingTime, timeoutCancellationTokenSource.Token));
 
+        if (completedTask.IsFaulted)
+        {
+            // Throw the inner exception.
+            await completedTask;
+        }
+
         if (completedTask != target)
         {
+            // The monitored task did not complete.
             return false;
         }
 
-        // cancel the clock
+        // The monitored task is completed, we shall cancel the clock.
         timeoutCancellationTokenSource.Cancel();
```

Comment:
🏕️ This name is confusing. Would you mind renaming it to something that better describes you're canceling the delay?